vault/eventbus/bus.go

```diff
@@ -217,9 +226,51 @@ func (bus *EventBus) Subscribe(ctx context.Context, ns *namespace.Namespace, pat
 	// add info needed to cancel the subscription
 	asyncNode.pipelineID = eventlogger.PipelineID(pipelineID)
 	asyncNode.cancelFunc = cancel
+	go bus.replayEvents(asyncNode)
 	return asyncNode.ch, asyncNode.Close, nil
 }
 
+func (bus *EventBus) copyRecentEvents() []*logical.EventReceived {
+	bus.recentEventsLock.RLock()
+	defer bus.recentEventsLock.RUnlock()
+
+	events := make([]*logical.EventReceived, bus.recentEvents.Len(), bus.recentEvents.Len())
+	for i := 0; i < bus.recentEvents.Len(); i++ {
+		events[i] = bus.recentEvents.At(i)
+	}
+	return events
+}
+
+func (bus *EventBus) replayEvents(node *asyncChanNode) {
+	//  we don't want to hold the lock while we send to the channel, so copy the event pointers first
+	events := bus.copyRecentEvents()
+	bus.logger.Info(fmt.Sprintf("Replaying %d events for %v subscriber", len(events), node.pipelineID))
+	for _, event := range events {
+		keep := true
+		if node.filter != nil {
+			var err error
+			keep, err = node.filter.Predicate(&eventlogger.Event{Payload: event})
+			if err != nil {
+				bus.logger.Warn("Error filtering event", "error", err)
+			}
+		}
+		if keep {
+			node.ch <- event
```

Comment:
I think we need to add the `cloudEventsFormatterFilter` node to this processing?